src/lib/Libnet/hnls.c

```diff
@@ -568,3 +570,140 @@ free_if_info(struct log_net_info *ni)
 		free(temp);
 	}
 }
+
+/**
+* @brief
+	Get a list of all IPs
+*
+* @return 
+*		Comma separated list of ips in string format
+* 
+* @par Side Effects:
+*	None
+*
+* @par MT-safe: Yes
+*
+*/
+char *
+get_all_ips(char *msg)
+{
+	char * nodenames = NULL;
+#if defined(linux)
+	char *tmp;
+	int len, hlen, ret;
+	struct ifaddrs *ifp, *listp;
+	char buf[80] = {'\0'};
+	char *p;
+
+	msg[0] = '\0';
+
+	ret = getifaddrs(&ifp);
+
+	if ((ret != 0) || (ifp == NULL)) {
+		strncpy(msg, "Failed to obtain interface names", 2048);
```

Comment:
Please refactor the hnls.c code a bit and make it nicer. For example, here there is hard coded length of 2048. Please use a  macro.

Reply:
I have added the necessary macro and refactored the hnls.c. Please continue further reviews. Thanks